Вам доступна таблица некоторых синтетических данных, на основании которых необходимо выполнить классификацию нового объекта, с помощью метода k-ближайших соседей.

In [1]:
import pandas as pd
import numpy as np
from sklearn.neighbors import KNeighborsClassifier
from sklearn.model_selection import train_test_split

In [2]:
data_loc = 'data.csv'
data = pd.read_csv(data_loc)
data.head()

id   X   Y  Class
0   1  81  83      1
1   2  53  33      1
2   3  10  69      1
3   4  52  78      0
4   5  43  79      0

In [3]:
cols = ['X','Y']
target = ['Class']
X = data[cols].loc[:]
y = data[target].loc[:]

In [4]:
X.head()

X   Y
0  81  83
1  53  33
2  10  69
3  52  78
4  43  79

In [5]:
y.head()

Class
0      1
1      1
2      1
3      0
4      0

In [14]:
neigh = KNeighborsClassifier(n_neighbors = 3, p = 2)
neigh.fit(X, y)

/usr/local/lib/python3.10/dist-packages/sklearn/neighbors/_classification.py:215: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples,), for example using ravel().
  return self._fit(X, y)


KNeighborsClassifier(n_neighbors=3)

In [15]:
test_data = pd.DataFrame({'X':[67], 'Y': [36]})

In [16]:
print(neigh.predict(test_data))

[1]


In [17]:
neigh.kneighbors(test_data, return_distance = True)

(array([[13.60147051, 14.31782106, 15.29705854]]), array([[6, 1, 9]]))

In [18]:
neigh = KNeighborsClassifier(n_neighbors = 3, p = 1)
neigh.fit(X, y)

/usr/local/lib/python3.10/dist-packages/sklearn/neighbors/_classification.py:215: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples,), for example using ravel().
  return self._fit(X, y)


KNeighborsClassifier(n_neighbors=3, p=1)

In [19]:
test_data = pd.DataFrame({'X':[67], 'Y': [36]})

In [20]:
print(neigh.predict(test_data))

[1]


In [21]:
neigh.kneighbors(test_data, return_distance = True)

(array([[17., 18., 19.]]), array([[1, 9, 6]]))

Введите расстояние от нового объекта с координатами (67,36) до ближайшего соседа, используя евклидову метрику: 13.60147051

Введите идентификатры трех ближайших точек к (67,36) для евклидовой метрики: 6, 1, 9

Введите класс для нового объекта с координатами (67,36) при k=3 и евклидовой метрике: 1

Введите расстояние от нового объекта с координатами (67,36) до ближайшего соседа, используя метрику городских кварталов (Манхеттенское расстояние): 17

Введите идентификатры трех ближайших точек к (67,36) для метрики городских кварталов: 1, 9, 6

Введите класс для нового объекта с координатами (67,36) при k=3 и метрике городских кварталов: 1